raw_section_snooper_vocab.py

```diff
@@ -0,0 +1,138 @@
+        concept_query = f"concept_code == '{code}' and vocabulary_id == '{vocabulary_id}'"
+        concept_row = concept_df.query(concept_query)
+
+        # If a match is found, return the concept name; otherwise, return 'n/a'
+        if concept_row.size > 1:
+            return concept_row['concept_name'].values[0]
+        return 'n/a'
+    return 'n/a'
+
+def snoop_for_code_tag(tree, expr, ns,concept_df):
+    """
+    Finds all elements matching the XPath expression (expr) in the XML tree and extracts relevant attributes.
+    Appends the extracted information to the global vocab_codes DataFrame.
+    """
+    section_elements = tree.findall(expr, ns)
+
+    # Use the global vocab_codes DataFrame
+    global vocab_codes
+
+    # Loop through each element matching the expression
+    for section_element in section_elements:
+        
+        # Extract relevant attributes from the document
+        tag = re.sub(r'{.*}', '', section_element.tag)
+        code = section_element.get('code')
+        codeSystem = section_element.get('codeSystem')
+        codeSystemName = section_element.get('codeSystemName')
+        displayName = section_element.get('displayName')
+        templateId = section_element.get('templateId')
+
+        # Print extracted details for debugging or logging purposes
+        print((f"\nSECTION  {type(section_element)} "
+               f" tag:{tag} "
+               f" code: {code} "
+               f" codeSystem: {codeSystem} "
+               f" codeSystemName: {codeSystemName} "
+               f" displayName: {displayName} "
+               f" templateId: {templateId}"))
+
+        # Create a new DataFrame row
+        new_row = pd.DataFrame([{
+            'tag': tag,
+            'code': code,
+            'codeSystem': codeSystem,
+            'codeSystemName': codeSystemName,
+            'displayName': displayName,
+            'templateId': templateId
+        }])
+
+        # Append the new row to vocab_codes using pd.concat for better performance
+        vocab_codes = pd.concat([vocab_codes, new_row], ignore_index=True)
+
+def main():
+    """
+    Main function that parses arguments, processes the XML file, extracts code elements, and cleans up the DataFrame.
+    """
+    parser = argparse.ArgumentParser(
+        prog='CCDA - OMOP Code Snooper',
+        description="Finds all code elements and shows what concepts they represent",
+        epilog='epilog?')
+    parser.add_argument('-f', '--filename', required=True, help="Filename of the XML file to parse")
+    args = parser.parse_args()
+
+    print("Reading Vocabulary, this may take a minute...")
+    concept_df = vocab_maps.read_concept()  # Uncomment if you need to load a concept DataFrame
+
+    # Parse the XML file
+    tree = ET.parse(args.filename)
+
+    # Extract code elements (tag-based and attribute-based)
+    snoop_for_code_tag(tree, ".//code", ns,concept_df)
```

Comment:
I bought (and likely forgot) your point about snooping for codeSystem attributes instead of code elements, because you'll miss things like AdministrativeGenderCode....so drop this first line.